### PR TITLE
Cdocinitial

### PR DESCRIPTION
**Pull request by couchbase123**
first commit for review

### CODE REVIEW

**Review by RichardSmedley — COMMENTED** [review]
Thanks for this, @couchbase123 
I've made some tweaks - https://github.com/couchbase/docs-sdk-c/commit/5df1ad9cccc33b95f1fd1d7828ba0044c1fdd7ed

Please note that we normally put each snippet from a page in a related file in the examples dir, make the file runnable (all imports, connection string, etc.), and bring it into the adoc with an `include::` - take a look at the Java docs repo to see this in action.

I'll let @avsej comment on the code snippets before we merge.  :+1:

**Review by osfameron — COMMENTED** [review]
Little confusing to have a merge commit in the pull request, would be good to have some description about what is changed.

File: modules/howtos/pages/subdocument-operations.adoc [hunk 1]
```diff
@@ -60,6 +47,38 @@ The paths `name`, `addresses.billing.country` and `purchases.complete[0]` are al
...
+        std::size_t idx = 0;
+        for (const auto& entry : results.entries) {
+            std::cout << idx << ": path=\"" << paths[idx] << "\", ";
+            if (entry.status == LCB_SUCCESS) {
+                std::cout << "value=" << (entry.value.empty() ? "(no value)" : entry.value) << "\n";
+            } else {
+                std::cout << "code=" << lcb_strerror_short(entry.status) << "\n";
+            }
+            ++idx;
+            ----
```
Comment (osfameron) [inline modules/howtos/pages/subdocument-operations.adoc]: Indentation (of both C code and Asciidoc code block closing) looks skewed here.

File: modules/howtos/examples/cas.c [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+newval += new_item;
+newval += "]";
+return newval;
+}
+
+// Boilerplate for storing our initial list as '[]'
+static void
+store_initial_list(lcb_INSTANCE* instance, const std::string& id)
+{
+std::string initial_document{ "[]" };
```
Comment (osfameron) [inline modules/howtos/examples/cas.c:100]: indentation

Comment (osfameron) [reply modules/howtos/examples/cas.c:100]: (from line 100-132)

**Comment by osfameron** [comment]
Left a few comments on minor tidying of commit descriptions, indentation etc.
I'm not yet up to speed on the C/C++ code, but it looks similar to the coding style of existing examples.

**Comment by RichardSmedley** [comment]
Adapted and updated in https://github.com/couchbase/docs-sdk-c/pull/93

**Pull request closed without merging**